documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,932 @@
+the [SubackPacket](https://aws.github.io/aws-iot-device-sdk-js-v2/browser/interfaces/mqtt5.SubackPacket.html) is received.
+You should always check the reason codes of a SUBACK completion to determine if the subscribe operation actually succeeded.
+
+> [!NOTE]
+> If you try to subscribe with the v2 MQTT5 client to a topic that is not allowed by a policy, you do not get the connection
+> closed but instead receive a SUBACK with a reason code.
+
+#### Example of subscribing in the v1 SDK
+
+```typescript
+device
+    .on('message', function(topic, payload) {
+        console.log('message', topic, payload.toString());
+    });
+
+device.subscribe('my/topic');
+```
+
+#### Example of subscribing in the v2 SDK
+
+```typescript
+client.on("messageReceived", (eventData: mqtt5.MessageReceivedEvent) : void => {
+    // evntData wraps message and topic.
+    console.log("message: " + JSON.stringify(eventData.message));
+});
+
+const suback: SubackPacket = await client.subscribe({
+    subscriptions: [
+        { qos: mqtt5_packet.QoS.AtMostOnce, topicFilter: "my/topic" }
+    ]
+});
+```
+
+
+### Unsubscribe
+
+To unsubscribe from topic in the v1 SDK, you should provide one or multiple topics to the [unsubscribe](https://github.com/mqttjs/MQTT.js/blob/master/README.md#unsubscribe)
+operation. You can provide an optional callback that will be fired on receiving UNSUBACK packet.
+
+In the v2 SDK, the [unsubscribe](https://aws.github.io/aws-iot-device-sdk-js-v2/browser/classes/mqtt5.Mqtt5Client.html#unsubscribe)
+operation takes a description of the [UnsubscribePacket](https://aws.github.io/aws-iot-device-sdk-js-v2/browser/interfaces/mqtt5.UnsubscribePacket.html)
+you wish to send and returns a promise that resolves successfully with the corresponding [UnsubackPacket](https://aws.github.io/aws-iot-device-sdk-js-v2/browser/interfaces/mqtt5.UnsubackPacket.html)
+returned by the broker. The `UnsubscribePacket` can describe more than one topic. The promise is rejected with an error if
+anything goes wrong before the [UnsubackPacket](https://aws.github.io/aws-iot-device-sdk-js-v2/browser/interfaces/mqtt5.UnsubackPacket.html)
+is received. You should always check the reason codes of an UNSUBACK completion to determine if the unsubscribe operation
+actually succeeded.
+
+#### Example of unsubscribing in the v1 SDK
+
+```typescript
+device.unsubscribe("my/topic", function(err) {
+    console.log("Unsubscribed");
+});
+```
+
+#### Example of unsubscribing in the v2 SDK
+
+```typescript
+let unsuback = await client.unsubscribe({
+    topicFilters: [ "my/topic" ]
+});
+console.log('Unsuback result: ' + JSON.stringify(unsuback));
+```
+
+
+### Client Stop
+
+In the v1 SDK, the `end` method in the `device` class disconnects the client. The force parameter determines whether to close
+the connection immediately, or wait for in-flight messages to be sent.
```

Comment:
Can a v1 client that has had `end` called on it be restarted or do you need to instantiate a new one to establish a new connection? If a disconnected client cannot be reused, it might be worth noting here that the v2 can be started, stopped, and restarted. That could be useful for someone that wants to have a client that is not persistently connected.